modules/Bio/EnsEMBL/Production/Pipeline/Xrefs/Checksum.pm

```diff
@@ -33,6 +33,12 @@ sub run {
   $self->dbc()->disconnect_if_idle() if defined $self->dbc();
   my ($user, $pass, $host, $port, $source_db) = $self->parse_url($db_url);
   my $dbi = $self->get_dbi($host, $port, $user, $pass, $source_db);
+  if ($skip_download) {
+    my $sth = $dbi->prepare("select 1 from checksum_xref limit 1");
+    $sth->execute;
+    my $table_nonempty = $sth->fetchrow_array;
```

Comment:
Perhaps
return $sth->rows > 0;
or even shortcut with
return $sth->rows
since not zero is true, just to clean it up a bit.